src/connectors/impls/gcs/writer.rs

```diff
@@ -0,0 +1,136 @@
+use gouth::Token;
+use crate::connectors::prelude::{
+    Attempt, EventSerializer, Result, SinkAddr, SinkContext, SinkManagerBuilder, SinkReply, Url,
+};
+use crate::connectors::sink::Sink;
+use crate::connectors::utils::url::HttpsDefaults;
+use crate::connectors::{CodecReq, Connector, ConnectorBuilder, ConnectorConfig, ConnectorType, Context};
+use http_client::h1::H1Client;
+use http_client::HttpClient;
+use http_types::{Method, Request};
+use value_trait::ValueAccess;
+use tremor_pipeline::{ConfigImpl, Event};
+use tremor_value::Value;
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct Config {
+    #[serde(default="default_endpoint")]
+    endpoint: String,
+    #[serde(default="default_connect_timeout")]
+    #[allow(unused)] // fixme! use or remove
+    connect_timeout: u64,
+    // request_timeout: u64
+}
+
+fn default_endpoint() -> String {
+    "https://storage.googleapis.com/upload/storage/v1".to_string()
+}
+
+fn default_connect_timeout() -> u64 {
+    10_000_000_000
+}
+
+impl ConfigImpl for Config {}
+
+#[derive(Debug, Default)]
+pub(crate) struct Builder {}
+
+#[async_trait::async_trait]
+impl ConnectorBuilder for Builder {
+    fn connector_type(&self) -> ConnectorType {
+        ConnectorType("gcs_writer".into())
+    }
+
+    async fn build_cfg(
+        &self,
+        _alias: &str,
+        _config: &ConnectorConfig,
+        connector_config: &Value,
+    ) -> Result<Box<dyn Connector>> {
+        let config = Config::new(connector_config)?;
+
+        Ok(Box::new(GCSWriterConnector { config }))
+    }
+}
+
+struct GCSWriterConnector {
+    config: Config,
+}
+
+#[async_trait::async_trait]
+impl Connector for GCSWriterConnector {
+    async fn create_sink(
+        &mut self,
+        sink_context: SinkContext,
+        builder: SinkManagerBuilder,
+    ) -> Result<Option<SinkAddr>> {
+        let url = Url::<HttpsDefaults>::parse(&self.config.endpoint)?;
```

Comment:
this ideally would be checked in `build_cfg` already so we get an error as soon as possible